packages/@lwc/template-compiler/src/parser/index.ts

```diff
@@ -1016,6 +1070,77 @@ function parseForOf(
     return node;
 }
 
+function parseScopedSlotFragment(
+    ctx: ParserCtx,
+    parse5Elm: parse5.Element,
+    parse5ElmLocation: parse5.ElementLocation,
+    parent: ParentNode,
+    parsedAttr: ParsedAttribute
+): ScopedSlotFragment | undefined {
+    const slotDataAttr = parsedAttr.pick(ElementDirectiveName.SlotData);
+    if (!slotDataAttr) {
+        return;
+    }
+
+    if (!ctx.config.enableScopedSlots) {
+        ctx.throwOnNode(ParserDiagnostics.INVALID_OPTS_LWC_SLOT_DATA, slotDataAttr);
+    }
+
+    if (parse5Elm.tagName !== 'template') {
+        ctx.throwOnNode(ParserDiagnostics.SCOPED_SLOT_DATA_ON_TEMPLATE_ONLY, slotDataAttr);
+    }
+
+    // 'lwc:slot-data' cannot be combined with other directives on the same <template> tag
+    if (ctx.findInCurrentElementScope(ast.isElementDirective)) {
+        ctx.throwAtLocation(
+            ParserDiagnostics.SCOPED_SLOTDATA_CANNOT_BE_COMBINED_WITH_OTHER_DIRECTIVE,
+            ast.sourceLocation(parse5ElmLocation)
+        );
+    }
+
+    // <template lwc:slot-data> element should always be the direct child of a custom element
+    // The only exception is, a conditional block as parent
+    const parentCmp = ctx.findAncestor(
+        ast.isComponent,
+        ({ current }) => current && ast.isConditionalBlock(current)
+    );
+
+    if (!parentCmp) {
+        ctx.throwAtLocation(
+            ParserDiagnostics.INVALID_PARENT_OF_LWC_SLOT_DATA,
+            ast.sourceLocation(parse5ElmLocation)
+        );
+    }
+
+    const slotDataAttrValue = slotDataAttr.value;
+    if (!ast.isStringLiteral(slotDataAttrValue)) {
+        ctx.throwOnNode(ParserDiagnostics.SLOT_DATA_VALUE_SHOULD_BE_STRING, slotDataAttr);
+    }
+
+    // Extract name of slot if incase its a named slot
+    const slotAttr = parsedAttr.pick('slot');
+    let slotName: Literal | undefined;
+    // Prevent usage of the slot attribute with expression.
+    if (slotAttr) {
+        if (ast.isExpression(slotAttr.value)) {
+            ctx.throwOnNode(ParserDiagnostics.SLOT_ATTRIBUTE_CANNOT_BE_EXPRESSION, slotAttr);
```

Comment:
The coverage report picked up this code path as not having a unit test, could you add one for this?